backend/src/api/petition-signature/controllers/petition-signature.js

```diff
@@ -0,0 +1,104 @@
+"use strict";
+
+/**
+ *  petition-signature controller
+ */
+
+const { createCoreController } = require("@strapi/strapi").factories;
+
+const path = require("path");
+
+function joinPaths(...args) {
+    return path.posix.join(...args).replace(':/', '://');
+}
+
+module.exports = createCoreController(
+  "api::petition-signature.petition-signature",
+  ({ strapi }) => ({
+    async add(ctx) {
+      const { petitionId, name, email, questionAnswers } = ctx.request.body;
+      if (!petitionId || !name || !email) {
+        ctx.throw(400, "Missing parameters");
+        return;
+      }
+      const petition = await strapi
+        .query("api::petition.petition")
+        .findOne({ where: { id: { $eq: petitionId } } });
+      if (!petition) {
+        ctx.throw(404, "petition not found");
+        return;
+      }
+
+      const petitionSignature = await strapi
+        .service("api::petition-signature.petition-signature")
+        .generateSignature(petition, name, email, questionAnswers);
+      const result = await strapi
+        .service("api::petition-signature.petition-signature")
+        .create({ data: petitionSignature });
+      if (!result) {
+        ctx.throw(500, "error generating signature");
+        return;
+      }
+
+      const exists = await strapi.entityService.count(
+        "api::petition-signature.petition-signature",
+        {
+          filters: {
+            petition: petitionId,
+            email: { $eq: email },
+          },
+        }
+      );
+
+      if (exists > 1) {
```

Comment:
Moet dit niet > 0 zijn?

Reply:
Nee want de check is nadat de nieuwe al gemaakt is. Anders krijg je race conditions